File: 5_frontend_module/topic_2/f6.py
# from functools import reduce

# a = [1, 4, 5]

# res = reduce(lambda x, y: x + y, a)

# print(res)

def outer(x):
    def inner(y):
        return x + y
    
    return inner

addFive = outer(5)
print(addFive(3))

File: 5_frontend_module/topic_2/f6.js
// res = 0;
// var a = [1, 4, 7, 10, 5, 3];

// for (let i = 0; i < a.length; i++) {
//   res = res + a[i];
// }

var a = [1, 4, 7, 10, 5, 3];
var res = a.reduce((total, current) => total + current, 5);

console.log(res);
